airflow/providers/google/cloud/operators/vertex_ai/custom_job.py

```diff
@@ -539,6 +564,94 @@ def on_kill(self) -> None:
+            command=self.command,
+            container_uri=self.container_uri,
+            model_serving_container_image_uri=self.model_serving_container_image_uri,
+            model_serving_container_predict_route=self.model_serving_container_predict_route,
+            model_serving_container_health_route=self.model_serving_container_health_route,
+            model_serving_container_command=self.model_serving_container_command,
+            model_serving_container_args=self.model_serving_container_args,
+            model_serving_container_environment_variables=self.model_serving_container_environment_variables,
+            model_serving_container_ports=self.model_serving_container_ports,
+            model_description=self.model_description,
+            model_instance_schema_uri=self.model_instance_schema_uri,
+            model_parameters_schema_uri=self.model_parameters_schema_uri,
+            model_prediction_schema_uri=self.model_prediction_schema_uri,
+            parent_model=self.parent_model,
+            is_default_version=self.is_default_version,
+            model_version_aliases=self.model_version_aliases,
+            model_version_description=self.model_version_description,
+            labels=self.labels,
+            training_encryption_spec_key_name=self.training_encryption_spec_key_name,
+            model_encryption_spec_key_name=self.model_encryption_spec_key_name,
+            staging_bucket=self.staging_bucket,
+            # RUN
+            dataset=Dataset(name=self.dataset_id) if self.dataset_id else None,
+            annotation_schema_uri=self.annotation_schema_uri,
+            model_display_name=self.model_display_name,
+            model_labels=self.model_labels,
+            base_output_dir=self.base_output_dir,
+            service_account=self.service_account,
+            network=self.network,
+            bigquery_destination=self.bigquery_destination,
+            args=self.args,
+            environment_variables=self.environment_variables,
+            replica_count=self.replica_count,
+            machine_type=self.machine_type,
+            accelerator_type=self.accelerator_type,
+            accelerator_count=self.accelerator_count,
+            boot_disk_type=self.boot_disk_type,
+            boot_disk_size_gb=self.boot_disk_size_gb,
+            training_fraction_split=self.training_fraction_split,
+            validation_fraction_split=self.validation_fraction_split,
+            test_fraction_split=self.test_fraction_split,
+            training_filter_split=self.training_filter_split,
+            validation_filter_split=self.validation_filter_split,
+            test_filter_split=self.test_filter_split,
+            predefined_split_column_name=self.predefined_split_column_name,
+            timestamp_split_column_name=self.timestamp_split_column_name,
+            tensorboard=self.tensorboard,
+        )
+        custom_container_training_job_obj.wait_for_resource_creation()
```

Comment:
Is it possible to do it in triggerer asynchronously?

Reply:
This call normally takes fractions of a second.  I put it here to make sure that all underlying resources are created before querying the name attribute of the Custom Container Training Job object, which corresponds to the training pipeline ID. I needed this pipeline ID for generating a link to the training pipeline **before** deferring the task.

Reply:
It does impact the event loop - so while it is ok to have it - maybe we could turn it into asyncio-compatible? I generally approve that one, but maybe extracting it into asyncio call would be a good idea?

Reply:
@potiuk I am not sure I understand what event loop is impacted. I'll try to explain my thinking below.

This `wait_for_resource_creation()` method belongs to the `CustomContainerTrainingJob` client class that is used for creating and launching a Vertex AI training pipeline. I call the method because it is recommended to do so  [here](https://github.com/googleapis/python-aiplatform/blob/main/google/cloud/aiplatform/base.py#L1376-L1405) as the next step after launching a training pipeline with parameter `sync=False`.  Despite its name, the method is not asynchronous and is not part of any event loops AFAIK. 

On the other hand, the existing trigger event loop uses another client class -  `PipelineServiceAsyncClient` to query the training pipeline status until its completion. It is not tied to `wait_for_resource_creation()` directly. This process can take a long time and that is why I chose to defer at this point.

To sum up:
- We launch a training pipeline synchronously by calling CustomContainerTrainingJob's `run()` and   `wait_for_resource_creation()` methods. This whole operation does not take more than a minute.
- We use CustomContainerTrainingJob's attribute for generating a Vertex AI link for the users.
- Then we defer, freeing up the worker slot, and wait until the training pipeline is completed using PipelineServiceAsyncClient's API. This operation could take hours.

Reply:
Ah. yeah. I just saw more context now, then yes - absolutely no problem with it tin this case

Reply:
@Lee-W ? What do you think?  Left the other comment for you to resolve

Reply:
Yep, sounds good!